### PR TITLE
Replace No data error with help message

### PR DESCRIPTION
**Pull request by etr2460**
### CATEGORY

Choose one

- [ ] Bug Fix
- [x] Enhancement (new features, refinement)
- [ ] Refactor
- [ ] Add tests
- [ ] Build / Development Environment
- [ ] Documentation

### SUMMARY
This PR updates the behavior of viz.py to not return an error when the results of a viz query returns 0 rows. It makes use of the update to SuperChart for handling this on the client side.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
Before:
Screen Shot 2020-03-05 at 2 26 11 PM

After:
Screen Shot 2020-03-04 at 12 48 56 PM

### TEST PLAN
test all viz types with a query that returns 0 rows, ensure the error message is shown and there aren't any js errors.

test all viz types with a query that returns some number of rows, and ensure the error message isn't shown

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [x] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### REVIEWERS
to: @john-bodley @kristw @mistercrunch @villebro

### CODE REVIEW

File: superset/viz.py [hunk 1]
```diff
@@ -1178,7 +1190,7 @@ def to_series(self, df, classed="", title_suffix=""):
             chart_data.append(d)
         return chart_data
 
-    def process_data(self, df, aggregate=False):
+    def process_data(self, df: pd.DataFrame, aggregate=False):
```
Comment (john-bodley) [inline superset/viz.py]: Nit. Whilst you here could you type the `aggregate` argument and return type?

**Review by john-bodley — APPROVED** [review]
LGMT.

**Review by kristw — APPROVED** [review]
Approve the js part

File: superset/viz.py (hunk 1 above)
Comment (etr2460) [reply superset/viz.py]: done

**Comment by etr2460** [comment]
Note, before merging I want to get https://github.com/apache-superset/superset-ui/pull/309 in and bump the chart package version again

File: superset/viz.py [hunk 2]
```diff
@@ -1907,7 +1936,7 @@ def get_df(self, query_obj: Optional[Dict[str, Any]] = None) -> pd.DataFrame:
         return pd.DataFrame()
 
     def get_data(self, df: pd.DataFrame) -> VizData:
-        return {}
```
Comment (robdiciuccio) [inline superset/viz.py]: Is this a separate issue?

**Review by robdiciuccio — APPROVED** [review]
LGTM

File: superset/viz.py (hunk 2 above)
Comment (etr2460) [reply superset/viz.py]: Context was discussed in the meetup; adding here...

If data is null or empty on the client side we render the `no results` message. iframes should never have data, but should still render the base chart, so i'm adding some dummy data here to ensure iframe charts render properly

**Pull request merged**